linux_os/guide/services/sssd/sssd_enable_pam_services/bash/shared.sh

```diff
@@ -3,17 +3,16 @@
 # Include source function library.
 . /usr/share/scap-security-guide/remediation_functions
 
-SSSD_SERVICES_PAM_REGEX="^[[:space:]]*\[sssd]([^\n]*\n+)+?[[:space:]]*services.*pam.*$"
-SSSD_SERVICES_REGEX="^[[:space:]]*\[sssd]([^\n]*\n+)+?[[:space:]]*services.*$"
-SSSD_PAM_SERVICES="[sssd]
-services = pam"
 SSSD_CONF="/etc/sssd/sssd.conf"
+SSSD_CONF_DIR="/etc/sssd/conf.d/*.conf"
 
-# If there is services line with pam, good
-# If there is services line without pam, append pam
-# If not echo services line with pam
-grep -q "$SSSD_SERVICES_PAM_REGEX" $SSSD_CONF || \
-	grep -q "$SSSD_SERVICES_REGEX" $SSSD_CONF && \
-	sed -i "s/$SSSD_SERVICES_REGEX/&, pam/" $SSSD_CONF || \
-	echo "$SSSD_PAM_SERVICES" >> $SSSD_CONF
+for f in $( ls $SSSD_CONF $SSSD_CONF_DIR 2> /dev/null ) ; do
+	# finds all services entries under [sssd] configuration category, get a unique list so it doesn't add redundant fix
+	services_list=$( awk '/^\s*\[/{f=0} /^\s*\[sssd\]/{f=1}f' $f | grep -P '^services[ \t]*=' | uniq )
 
+	while IFS= read -r services; do
+		if [[ ! $services =~ "pam" ]]; then
+			sed -i "s/$services$/&, pam/" $f
+		fi
+	done <<< "$services_list"
```

Comment:
What happens if there are two identical `services` lines?
I guess only one of the lines has `pam` appended.

Reply:
both of the lines will have `pam` appended. That's my understanding of this `sed` command, every occurrence matching the `value` will have `pam` appended.

Reply:
if these two lines are under the same `[sssd]` section, only the last line of `services` will be considered by the OVAL check, but if they are somehow part of different `[sssd]` sections, OVAL will test both of them.

Reply:
@ggbecker You are correct, :)